ipympl/backend_nbagg.py

```diff
@@ -306,28 +308,36 @@ def _repr_mimebundle_(self, **kwargs):
 
         buf = io.BytesIO()
         self.figure.savefig(buf, format='png', dpi='figure')
-        self._data_url = b64encode(buf.getvalue()).decode('utf-8')
-        # Figure width in pixels
+
+        base64_image = b64encode(buf.getvalue()).decode('utf-8')
+        self._data_url = f'data:image/png;base64,{base64_image}'
+        # Figure size in pixels
         pwidth = self.figure.get_figwidth() * self.figure.get_dpi()
+        pheight = self.figure.get_figheight() * self.figure.get_dpi()
         # Scale size to match widget on HiDPI monitors.
         if hasattr(self, 'device_pixel_ratio'):  # Matplotlib 3.5+
             width = pwidth / self.device_pixel_ratio
+            height = pheight / self.device_pixel_ratio
         else:
             width = pwidth / self._dpi_ratio
+            height = pheight / self._dpi_ratio
         html = """
             <div style="display: inline-block;">
                 <div class="jupyter-widgets widget-label" style="text-align: center;">
                     {}
                 </div>
-                <img src='data:image/png;base64,{}' width={}/>
+                <img src='{}' width={}/>
             </div>
         """.format(
             self._figure_label, self._data_url, width
         )
 
+        # Update the widget model properly for HTML embedding
+        self._size = (width, height)
```

Comment:
This is one important change here.

We need to make the plot size part of the widget model in the case of `nbconvert --to html` otherwise there is no registered size and the canvas would be 0 sized.